Core/BookmarksCoreDataStorage.swift

```diff
@@ -135,6 +124,28 @@ public class BookmarksCoreDataStorage {
             }            
         }
     }
+
+    static internal func rootFolderManagedObject(_ context: NSManagedObjectContext) -> BookmarkFolderManagedObject {
+        guard let bookmarksFolder = NSEntityDescription.insertNewObject(forEntityName: "BookmarkFolderManagedObject", into: context)
+                as? BookmarkFolderManagedObject else {
+            // TODO - this should not be fatal
```

Comment:
TODO

Reply:
please see: https://app.asana.com/0/0/1201927187914110/f

Reply:
I’ve removed the TODO; I had left it as a reminder / question for @brindy about fatalError in this particular scenario

@jaceklyp re: the CoreData insert that’s legacy code that I encapsulated into a method only for reusability and to make it unit testable. I don’t think I should make that change in this PR as I don’t know what else in the app it can impact / break and don’t know the test scenarios. The core data change should be done in a different PR I think.

Here is a screenshot of what code was before and after 
<img width="1118" alt="image" src="https://user-images.githubusercontent.com/91189922/161255704-a045e522-81e3-4203-8c89-a2c5220e44df.png">
<img width="922" alt="image" src="https://user-images.githubusercontent.com/91189922/161255713-b89fddbb-39c8-4256-8d38-19195bbf619e.png">

Reply:
It's really up to you :) Personally I prefer to fix the broken code when I see one and this is really a syntactic sugar change from:
`guard let bookmarksFolder = NSEntityDescription.insertNewObject(forEntityName: "BookmarkFolderManagedObject", into: context)
                as? BookmarkFolderManagedObject else {`
to:
`let bookmarksFolder: BookmarkFolderManagedObject = context.insertObject()`

And it helps with your issue because then you don't have to use fatalErrors anymore, because the code is safely unwrapped for you and it doesn't have any negative effect on anything else - meaning - just checking if this code works is sufficient enough.